ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/replyboxview/ReplyBoxFragmentTest.java

```diff
@@ -0,0 +1,350 @@
+
+    @Mock
+    private EditText editText;
+
+    @Mock
+    private TextView textView;
+
+    @Mock
+    private KeyEvent keyEvent;
+
+    @Mock
+    private Intent intent;
+
+    @Mock
+    private AppCompatActivity appCompatActivity;
+
+    @Mock
+    private Editable editable;
+
+    @Mock
+    private ImageButton imageButton;
+
+    @Mock
+    private ReplyBoxContract.ReplyBoxListener listener;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Integer> integerArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Boolean> booleanArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<ReplyBoxContract.ReplyBoxListener> replyBoxListenerArgumentCaptor;
+
+    @Rule
+    public final ErrorCollector errorCollector = new ErrorCollector();
+
+    @Test
+    public void onCreate() {
+        //Act
+        replyBoxFragment.onCreate(bundle);
+        final ReplyBoxContract.Presenter mPresenter =
+                Whitebox.getInternalState(replyBoxFragment, "mPresenter");
+
+        //Assert
+        assertNotNull(mPresenter);
+    }
+
+    @Test
+    public void onCreateView() {
+        //Arrange
+        when(layoutInflater.inflate(R.layout.ko__fragment_reply_box, viewGroup, false)).thenReturn(view);
+
+
+        //Act
+        replyBoxFragment.onCreateView(layoutInflater, viewGroup, bundle);
+        final View mRoot = Whitebox.getInternalState(replyBoxFragment, "mRoot");
+
+        //Assert
+        assertEquals(view, mRoot);
+    }
+
+    @Test
+    public void onViewCreated() throws Exception {
+        //Arrange
+        final String dummyPowerMessage = "dummyPowerMessage";
+        suppress(methods(Fragment.class, "startActivity"));
+        mockStatic(KayakoCreditsHelper.class);
+        mockStatic(Uri.class);
+        uri = mock(Uri.class);
+        Whitebox.setInternalState(replyBoxFragment, "mRoot", view);
+        Whitebox.setInternalState(replyBoxFragment, "mPresenter", presenter);
+        when(view.findViewById(R.id.ko__reply_box_send_button)).thenReturn(view);
+        when(view.findViewById(R.id.reply_box_edittext)).thenReturn(editText);
+        when(keyEvent.getKeyCode()).thenReturn(KeyEvent.KEYCODE_ENTER);
+        when(view.findViewById(R.id.ko__reply_box_attach_button)).thenReturn(view);
+        when(view.findViewById(R.id.ko__reply_box_credits_by)).thenReturn(textView);
+        when(KayakoCreditsHelper.getPoweredByMessage()).thenReturn(dummyPowerMessage);
+        when(view.findViewById(R.id.ko__reply_box_credits_kayako)).thenReturn(textView);
+        when(KayakoCreditsHelper.getLink()).thenReturn("link");
+        when(Uri.parse("link")).thenReturn(uri);
+        whenNew(Intent.class).withArguments(Intent.ACTION_VIEW, uri).thenReturn(intent);
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                Object arguments[] = invocation.getArguments();
```

Comment:
⚠️  Array designators should always be located on the type for better code readability. Otherwise, developers must look both at the type and the variable name to know whether or not a variable is an array.

**[More details](https://confluence.devfactory.com/x/zhCUEw)**
          
## The possible solution in this case
```Java
Object[] arguments = invocation.getArguments();
``` 

<a id="15ead1d0-53ca-11e8-b777-83ccc384eaae" href="none" codereview-uuid-reserved-tag />